src/tests/runTest.ts

```diff
@@ -34,7 +49,7 @@ async function main() {
     // Passed to --extensionTestsPath
     const extensionTestsPath = path.resolve(__dirname, './suite/index')
 
-    const vscodeExecutablePath = await downloadAndUnzipVSCode('1.68.0')
+    const vscodeExecutablePath = await downloadAndUnzipVSCode('stable')
```

Comment:
This is where we'd get a self-signed certificate in the certificate chain error when on the company VPN.

Reply:
Good question Steve.

For testing the extension, I think we ought to be floating on the latest stable, which this patch facilitates by using `'stable'` as the version instead of 1.68.0, which is pretty old now.  With 1.68.0, I was getting the following errors, which are fixed when using `'stable'`:

```
Extension is not compatible with Code 1.68.0. Extension requires: ^1.74.0.
Extension is not compatible with Code 1.68.0. Extension requires: ^1.74.2.
Extension is not compatible with Code 1.68.0. Extension requires: ^1.74.0.
```

Now pinning the version does allow us to _cache_ a local copy and not have to issue an HTTPS request to discover the current stable version, which was a problem for us before.  Since we were pinning to 1.68.0, we could hop off the VPN, download the artifact, then sign back into the VPN and continue testing without any HTTPS traffic.

Reply:
Pinning feels perferable to me:

1. Disabling https doesn't feel great to me if it can be easily avoided.
2. Pinning to a particular version does help us to better define a minimum support version, or at least a known good version.
3. If a VS update breaks something, it won't immediately break all of our tests and gives time to figure out what changes are needed to succesfully update. Then we can pin to a certain version

Also, if this version came from an environment variable or something we could potentially have CI run with different VS Code versions if that was something that was of value.

Reply:
You make some valid points and appreciate this discussion.

You always want to pin your dependencies so you know precisely what is in your kit. This is a little different though as the extension is expected to run inside the users application where we have little control (we can have a "known good") over the version they have installed.  I imagine most users are running close to the latest stable with VS Code checking for updates and just the ease in which the update system works.  If the latest stable is breaking us, I think we ought to know as soon as practical because it _will_ affect our users and pinning it down to some old version will not allow us to "get out in front" of a potential issue, making us appear to be "sleeping behind the wheel".  That said, I like the idea of externalizing the version choice so we can test various versions as needed.  Perhaps a hybrid approach were we use a pinned "known good" version and also test the latest stable.  What factors shall we be considering when pinning down a VS Code version for testing?  Perhaps at the beginning of a release cycle we select the most recent stable and document that as the "known good" while also testing on the latest stable?

Reply:
> If the latest stable is breaking us, I think we ought to know as soon as practical

I agree, but defaulting to stable has some issues. For example, say VS Code makes an update to stable that breaks our extension. We can fix it then, but now if we ever try to run tests in a older commit that uses stable (e.g. to git bisect a bug) all those old tests will fail. If someone wants to run old tests, they should run exactly the same as it did back when those tests were committed. Version pinning is the right way to do this.

>  Perhaps a hybrid approach were we use a pinned "known good" version and also test the latest stable. 

Yeah, I think is is a good idea. I don't know enough about VS code to know what version to pin to or how often new releases are made. I think your suggestion is reasonable, but I'll leave it to the VS Code experts to decie.

I would suggest we create a GitHub action that runs our tests nightly and runs a test with stable (or maybe event nightly releases?) using your new environment variable and sends out some notification if it fails. This way tests are immutable and don't change depending on what stable happens to be at the time of running, but we also stay on top of changes to VS code and make sure our extension continues to work with new releases.

Reply:
Pinning done via CI.yaml using the `DAFFODIL_TEST_VSCODE_VERSION` environment variable.  Docs added for testing against a specific version of VS Code: https://github.com/apache/daffodil-vscode/wiki/Apache-Daffodil™-Extension-for-Visual-Studio-Code:-Development#testing-against-a-specific-version-of-vs-code